src/jobBundle/Controller/JobController.php

```diff
@@ -214,15 +213,15 @@ public function getJobsForDelete() {
      * @Route("job/job_delete/{job_id}")
      */
     public function deleteJob($job_id) {
-        if (!$this->jobService->findJob($job_id)) {
+        if ($this->jobService->retriveJobById($job_id) == null) {
             // If the id doesn't exist in the database
             return $this->render(
                             'job/job_error.html.twig', array(
                         'data' => $job_id
             ));
         } else {
             // Do the business logic
-            $this->jobService->deleteJob($this->jobService->getJobName($job_id));
+            $this->jobService->removeById($this->jobService->retriveJobById($job_id));
```

Comment:
If removeJobById does what it says then you don't need to retrieve the entity by id. ;)